internal/authentication/password_hash_test.go

```diff
@@ -7,49 +7,122 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestShouldHashPassword(t *testing.T) {
-	hash := HashPassword("password", "$6$rounds=50000$aFr56HjK3DrB8t3S")
+func TestShouldHashSHA512Password(t *testing.T) {
+	hash, err := HashPassword("password", "aFr56HjK3DrB8t3S", HashingAlgorithmSHA512, 50000, 0, 0, 16)
+	assert.NoError(t, err)
 	assert.Equal(t, "$6$rounds=50000$aFr56HjK3DrB8t3S$zhPQiS85cgBlNhUKKE6n/AHMlpqrvYSnSL3fEVkK0yHFQ.oFFAd8D4OhPAy18K5U61Z2eBhxQXExGU/eknXlY1", hash)
 }
 
-func TestShouldCheckPassword(t *testing.T) {
+func TestShouldHashArgon2idPassword(t *testing.T) {
+	hash, err := HashPassword("password", "BpLnfgDsc2WD8F2q", HashingAlgorithmArgon2id, 3, 65536, 2, 16)
+	assert.NoError(t, err)
+	assert.Equal(t, "$argon2id$v=19$m=65536,t=3,p=2$BpLnfgDsc2WD8F2q$o/vzA4myCqZZ36bUGsDY//8mKUYNZZaR0t4MFFSs+iM", hash)
+}
+
+func TestShouldNotHashPassword(t *testing.T) {
+	hash, err := HashPassword("password", "BpLnfgDsc2WD8F2q", "bogus", 3, 65536, 2, 16)
+	assert.Equal(t, "", hash)
+	assert.EqualError(t, err, "Hashing Algorithm 'bogus' is Invalid (only support values of argon2id and 6).")
+}
+
+func TestShouldNotHashArgon2idPasswordDueToMemoryParallelismMismatch(t *testing.T) {
```

Comment:
Add tests to also cover `memory < 8` and `parallelism < 1`

Reply:
(side note) I'll check what kind of tool we could consider using for code coverage in the future. It would be helpful here for instance.

Reply:
I actually did that previously, somehow lost the change. I may have inadvertently discarded it. Done locally.